dftimewolf/lib/processors/gcp_cloud_resource_tree.py

```diff
@@ -0,0 +1,1199 @@
+      elif resource_type == 'instances':
+        self.type = 'gce_instance'
+      elif resource_type == 'images':
+        self.type = 'gce_image'
+      elif resource_type == 'machineImages':
+        self.type = 'gce_machine_image'
+      elif resource_type == 'instanceTemplates':
+        self.type = 'gce_instance_template'
+      elif resource_type == 'snapshots':
+        self.type = 'gce_snapshot'
+      else:
+        self.type = resource_type
+
+      if '/zones/' in value:
+        self.zone = values[-3]
+        self.project_id = values[-5]
+      elif '/global/' in value:
+        self.zone = 'global'
+        self.project_id = values[-4]
+
+    self._resource_name = value
+
+  @property
+  def creation_timestamp(self) -> Optional[datetime]:
+    """Property creation_timestamp Getter."""
+    return self._creation_timestamp
+
+  @creation_timestamp.setter
+  def creation_timestamp(self, value: datetime) -> None:
+    """Property creation_timestamp Setter."""
+    if isinstance(value, datetime):
+      self._creation_timestamp = value
+    else:
+      self._creation_timestamp = parser.parse(value)
+
+  @property
+  def deletion_timestamp(self) -> Optional[datetime]:
+    """Property deletion_timestamp Getter."""
+    return self._deletion_timestamp
+
+  @deletion_timestamp.setter
+  def deletion_timestamp(self, value: datetime) -> None:
+    """Property deletion_timestamp Setter."""
+    if isinstance(value, datetime):
+      self._deletion_timestamp = value
+    else:
+      self._deletion_timestamp = parser.parse(value)
+
+  def IsDeleted(self) -> bool:
+    """Check if resource is deleted."""
+    if self.deleted or self.deletion_timestamp or not self.creation_timestamp:
+      return True
+
+    return False
+
+  def GenerateTree(self) -> List[Dict[str, 'Resource']]:
+    """Generate the resource tree.
+
+    Returns:
+      List of dictionaries containing a reference to the resource and it name
+      indented based on it's location in the tree
+    """
+    tab = '\t'
+    output: List[Any] = []
+
+    # Count the number of parent resources and the level in the tree the
+    # resource is at.
+    level = 0
+    parent_resource = self.parent
+    while parent_resource:
+      level = level + 1
+      parent_resource = parent_resource.parent
+
+    # Add the resource parent entries to the List of dictionaries
+    counter = level
+    parent_resource = self.parent
+    if parent_resource:
+      while counter > 0:
+        counter = counter - 1
+        entry: Dict[str, Any] = {}
+        entry['resource_object'] = parent_resource
+        entry['graph'] = f'{tab*counter}|--{parent_resource.name}'
+        output.insert(0, entry)
+        if parent_resource.parent:
+          parent_resource = parent_resource.parent
+
+    # Add resource entry to the List of dictionaries
+    entry = {}
+    entry['resource_object'] = self
+    entry['graph'] = f'{tab*level}|--{self.name}'
+    output.insert(level, entry)
```

Comment:
insert the current entry first to index zero in the list and then the children to have the same order of insertion 

```suggestion
    # Add current resource entry to the List of dictionaries
    entry = {}
    entry['resource_object'] = self
    entry['graph'] = f'{tab*level}|--{self.name}'
    output.insert(0, entry)
    
    # Add the resource parent entries to the List of dictionaries
    counter = level
    parent_resource = self.parent
    if parent_resource:
      while counter > 0:
        counter = counter - 1
        entry: Dict[str, Any] = {}
        entry['resource_object'] = parent_resource
        entry['graph'] = f'{tab*counter}|--{parent_resource.name}'
        output.insert(0, entry)
        if parent_resource.parent:
          parent_resource = parent_resource.parent
```

Reply:
Not sure what you want to achieve with that change. It seems to be the same for me.  I just add things in order they are printed: parent resource, resource in question and then children resource for things to be more readable.

Reply:
@sa3eed3ed please check if I should mark this as resolved.